common/src/main/scala/com/aliyun/emr/rss/common/internal/Logging.scala

```diff
@@ -32,142 +30,58 @@ trait Logging {
   }
 
-  // Method to get or create the logger for this object
-  protected def log: Logger = {
-    if (log_ == null) {
-      initializeLogIfNecessary(false)
-      log_ = LoggerFactory.getLogger(logName)
-    }
-    log_
-  }
-
   // Log methods that take only a String
-  protected def logInfo(msg: => String) {
-    if (log.isInfoEnabled) log.info(msg)
+  protected def logTrace(msg: => String): Unit = {
+    if (log.isTraceEnabled) log.trace(msg)
   }
 
-  protected def logDebug(msg: => String) {
+  protected def logDebug(msg: => String): Unit = {
     if (log.isDebugEnabled) log.debug(msg)
   }
 
-  protected def logTrace(msg: => String) {
-    if (log.isTraceEnabled) log.trace(msg)
+  protected def logInfo(msg: => String): Unit = {
+    if (log.isInfoEnabled) log.info(msg)
   }
 
-  protected def logWarning(msg: => String) {
+  protected def logWarning(msg: => String): Unit = {
     if (log.isWarnEnabled) log.warn(msg)
   }
 
-  protected def logError(msg: => String) {
+  protected def logError(msg: => String): Unit = {
     if (log.isErrorEnabled) log.error(msg)
   }
 
-  // Log methods that take Throwables (Exceptions/Errors) too
-  protected def logInfo(msg: => String, throwable: Throwable) {
-    if (log.isInfoEnabled) log.info(msg, throwable)
+  // Log methods that take Throwable(Exception/Error) too
+  protected def logTrace(msg: => String, throwable: Throwable): Unit = {
+    if (log.isTraceEnabled) log.trace(msg, throwable)
   }
 
-  protected def logDebug(msg: => String, throwable: Throwable) {
+  protected def logDebug(msg: => String, throwable: Throwable): Unit = {
     if (log.isDebugEnabled) log.debug(msg, throwable)
   }
 
-  protected def logTrace(msg: => String, throwable: Throwable) {
-    if (log.isTraceEnabled) log.trace(msg, throwable)
+  protected def logInfo(msg: => String, throwable: Throwable): Unit = {
+    if (log.isInfoEnabled) log.info(msg, throwable)
   }
 
-  protected def logWarning(msg: => String, throwable: Throwable) {
+  protected def logWarning(msg: => String, throwable: Throwable): Unit = {
     if (log.isWarnEnabled) log.warn(msg, throwable)
   }
 
-  protected def logError(msg: => String, throwable: Throwable) {
+  protected def logError(msg: => String, throwable: Throwable): Unit = {
     if (log.isErrorEnabled) log.error(msg, throwable)
   }
-
-  protected def isTraceEnabled(): Boolean = {
-    log.isTraceEnabled
-  }
-
-  protected def initializeLogIfNecessary(isInterpreter: Boolean): Unit = {
-    initializeLogIfNecessary(isInterpreter, silent = false)
-  }
-
-  protected def initializeLogIfNecessary(
-      isInterpreter: Boolean,
-      silent: Boolean = false): Boolean = {
-    if (!Logging.initialized) {
-      Logging.initLock.synchronized {
-        if (!Logging.initialized) {
-          initializeLogging(isInterpreter, silent)
-          return true
-        }
-      }
-    }
-    false
-  }
-
-  private def initializeLogging(isInterpreter: Boolean, silent: Boolean): Unit = {
```

Comment:
it's actually a dead code becuase
1) rss claims to support both log4j1 and log4j2, the code handles log4j1 only.
2) if user does not provide log4j1 configurations in classpath, it trying to load an un-existing `log4j-defaults.properties`